### ISSUE TITLE
Unzipping an SSZipArchive zip file loses executable permissions

### ISSUE THREAD

**Issue by andalman** [ZipArchive/ZipArchive#689]
### Steps to reproduce
1. Create a simple executable shell script
2. Zip it up using SSZipArchive
3. Unzip the archive by double-clicking it in the Finder

### Expected behavior
You should be able to run the simple executable shell script

### Actual behavior
You cannot run the shell script: it does not have executable permissions

### Version of ZipArchive
2.5.5

### Environmental information
I’m running this on macOS Ventura 13.6

### Additional information

I’ve attached an Xcode project that illustrates the issue. If you run it, two zip archives will be created in ~/Desktop:
archive-zipUnix.zip
archive-ZipArchive.zip

The archive with the “zipUnix” suffix is created by running the builtin macOS command line `zip` to create the zip file. Double-clicking in the Finder will create a “hello.sh” script that has executable permissions and can be run.

The archive with the “ZipArchive” suffix is created using SSZipArchive. Double-clicking in the Finder will create a “hello.sh” file without executable permissions.

If you run `zipinfo` on each, you’ll see the following:

```
[host] Desktop $ zipinfo -l archive-zipUnix.zip 
Archive:  archive-zipUnix.zip
Zip file size: 254 bytes, number of entries: 1
-rwxr-xr-x  3.0 unx       97 tx       88 defN 23-Oct-16 16:37 hello.sh
1 file, 97 bytes uncompressed, 88 bytes compressed:  9.3%

[host] Desktop $ zipinfo -l archive-ZipArchive.zip 
Archive:  archive-ZipArchive.zip
Zip file size: 266 bytes, number of entries: 1
-rwxr-xr-x  0.0 osx       97 bX       88 defN 23-Oct-16 16:37 hello.sh
1 file, 97 bytes uncompressed, 88 bytes compressed:  9.3%

```

Two obvious differences standout: the version for “hello.sh” in the archive created by `zip` is 3.0 and maintains it was created on “unx” (unix), while the version in the archive created by SSZipArchive is 0.0 (!) and maintains it was created on “osx”. 

If I hardcode the “version_madeby” to be UNIX in my personal build of SSZipArchive, everything works as I’d expect. I’m not sure that’s the best answer however.

[zipper.zip](https://github.com/ZipArchive/ZipArchive/files/12923225/zipper.zip)

**Comment by fvolchyok** [comment]
I'm not an expert on archive formats, but seems like zip does not guarantee support for file permissions? More info is here: https://superuser.com/a/1257441/758853

**Comment by Coeur** [comment]
The "version_madeby" is configured differently in two places

### When writing a file to an archive:
https://github.com/ZipArchive/ZipArchive/blob/be6ac340cbe394cc6a81eae6034a91a7702d9db3/SSZipArchive/SSZipArchive.m#L1326
That is coming from b1b58b99d61475ce26b85b2a92668d82958c4080 by me, @Coeur, with that comment:
>Fix version_madeby for unicode support

It was done to fix #507.

### When writing a symlink to an archive:
https://github.com/ZipArchive/ZipArchive/blob/be6ac340cbe394cc6a81eae6034a91a7702d9db3/SSZipArchive/SSZipArchive.m#L1023
That is coming from #560 by @willson-chen and reviewed by @jhudsonWA with that review comment:
>Only `3 << 8` is avaible to zip symlink. If it is changed to `19 << 8`, the symlink file will be zipped failed, which puzzles me.

**Comment by Coeur** [comment]
So we have three tasks:
- look into the executable permissions issue when version_madeby is UNIX (3) or OSX (19).
- look into unicode filename support when version_madeby is UNIX (3) or OSX (19).
- look into symlink support when version_madeby is UNIX (3) or OSX (19).

**Comment by Coeur** [comment]
First, I checked the doc:
https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT
>4.4.2.1 The upper byte indicates the compatibility of the file
        attribute information.

>4.4.2.3 The lower byte indicates the ZIP specification version 
        (the version of this document) supported by the software 
        used to encode the file.

Since that document is at version 6.3.10, and I don't know the history of the document, I may need to contact zipformat@pkware.com to get the earlier specs and figure out the proper version number that we support (instead of 0.0).

**Comment by Coeur** [comment]
https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT

>D.2 If general purpose bit 11 is unset, the file name and comment SHOULD conform 
to the original ZIP character encoding.  If general purpose bit 11 is set, the 
filename and comment MUST support The Unicode Standard, Version 4.1.0 or 
greater using the character encoding form defined by the UTF-8 storage 
specification.  The Unicode Standard is published by the The Unicode
Consortium (www.unicode.org).  UTF-8 encoded data stored within ZIP files 
is expected to not include a byte order mark (BOM). 

So for Unicode support, the correct fix was done in #564, and not by me in b1b58b99d61475ce26b85b2a92668d82958c4080.

So we don't have compelling reasons to keep using 19 (OSX) for now.

**Comment by andalman** [comment]
Thank you so much!
